Owncloud iOs Client/Login/Login/UniversalLoginViewController.swift

```diff
@@ -933,24 +923,31 @@ public enum TextfieldType: String {
                     
                 } else {
 
+                    //1.Generate correct user
+                    if self.loginMode == .create {
+                        self.user = UserDto()
+                    }
+                    
                     if self.loginMode == .create || self.loginMode == .migrate || self.forceAccountMigration {
-                        let newUser = UserDto()
+
                         if self.loginMode != .create {
-                            newUser.userId = self.user!.userId
-                            if self.loginMode == .migrate {
-                                newUser.predefinedUrl = k_default_url_server
-                            }
+                            self.userNewCredentials.userId = String(self.user!.userId)
                         }
-                        newUser.url = self.validatedServerURL
-                        newUser.username = self.userNewCredentials.userName
-                        newUser.ssl = self.validatedServerURL.hasPrefix("https")
-                        newUser.urlRedirected = app.urlServerRedirected
-                        
-                        self.user = newUser.copy() as? UserDto
+                        self.user!.url = self.validatedServerURL
+                        self.user!.username = self.userNewCredentials.userName
+                        self.user!.ssl = self.validatedServerURL.hasPrefix("https")
+                        self.user!.urlRedirected = app.urlServerRedirected
                     }
-
+                    
+                    self.user!.predefinedUrl = k_default_url_server
                     self.userNewCredentials.baseURL = UtilsUrls.getFullRemoteServerPath(self.user)
+                    
+                    if self.forceAccountMigration {
```

Comment:
If you use the suggestion provided in my previous comment you don't need to checks

Reply:
This refactor will imply to pass testcases of all different login modes and distinguish expired mode from force account in all the login view, right now forceAccountMigration it is a special expired mode. I would not suggest do it here,